### PR TITLE
AmcCarrier Core Python Restructuring

### PR DESCRIPTION
**Pull request by slacrherbst**
Instead of relying on an external python package called "common" this new method will rely on the developer adding an object which is a sub-class of AppTop.AppCore to the device tree from a higher level python script.

class MyAppCore(AppTop.AppCore):
    ....

self.FpgaTopLevel.AppTop.add(MyAppCore(name='blah'))

All of the AmcCarrierCore python modules are moved down one level below an AmcCarrierCore namespace.

This PR also removes the link details from the TopLevel module.

### CODE REVIEW

**Comment by bengineerd** [comment]
I started a python coding guide for rogue:
https://github.com/slaclab/rogue/wiki/Best-Practices
We should make sure this PR complies.

File: python/AmcCarrierCore/AppMps/_AppMps.py [hunk 1]
```diff
@@ -18,23 +18,24 @@
 #-----------------------------------------------------------------------------
 
 import pyrogue as pr
-import AppMps  as mps
+from AmcCarrierCore.AppMps._AppMpsSalt import AppMpsSalt
+from AmcCarrierCore.AppMps._AppMpsThr  import AppMpsThr
```
Comment (bengineerd) [inline python/AmcCarrierCore/AppMps/_AppMps.py]: Python modules that are named with a leading underscore should not be imported like this. Only __init__.py should import them.

**Comment by slacrherbst** [comment]
Please merge this.

**Comment by slacrherbst** [comment]
Ok. COnflicts resolved.

**Pull request merged**